rep-2004.rst

```diff
@@ -0,0 +1,681 @@
+You may either try to apply the requirements for the quality levels described here, or you may wish to simply argue the quality without using these requirements as a ruler.
+The argument could be text justification, or it could link to other analyses or discussions had by community members rationalizing the choice.
+
+For example, if your package depends on ``rclcpp``, and ``rclcpp`` claims 'Level 1' quality with the caveat that this requires you use an rmw implementation that also meets the 'Level 1' quality standard, then your package's "quality declaration" document should mention this as well.
+This could be accomplished simply by saying that one of your dependencies, ``rclcpp``, has some caveats and then linking to ``rclcpp``'s own "quality declaration".
+
+There's obviously a lot of ambiguity in this area, as you could argue for or against a variety of dependencies and how they affect the quality of a package.
+The goal is for the maintainers of a package to "make the case" that their dependencies are at least as high quality as the package in question.
+They should examine each dependency, and document any important caveats or justified exceptions in the package's "quality declaration" document so peer reviewers and consumers of the package can make their own evaluations.
+
+Quality Level 2
+^^^^^^^^^^^^^^^
+
+These are packages which need to be solidly developed and might be used in production environments, but are not strictly required, or are commonly replaced by custom solutions.
+This can also include packages which are not yet up to 'Level 1' but intend to be in the future.
+
+Package Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+Requirements to be considered a 'Level 2' package:
+
+#. **Version Policy:**
+
+   #. The same as 'Level 1' packages
+
+#. **Change Control Process:**
+
+   #. Must have all code changes occur through a change request (e.g. pull request, merge request, etc.)
+   #. Must have confirmation of contributor origin (e.g. `DCO  <https://developercertificate.org/>`_, CLA, etc.)
+   #. Must have Continuous Integration (CI) policy for all change requests
+
+#. **Documentation:**
+
+   #. Must have documentation for each "feature" (e.g. for ``rclcpp``: create a node, publish a message, spin, etc.)
+   #. Must have a declared license or set of licenses
+   #. Must have a copyright statement in each source file
+   #. Must have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements
+
+      #. *Must have a section in the repository's ``README`` which contains the "quality declaration" or links to it*
+      #. *Must register with a centralized list of 'Level 2' packages, if one exists, to allow for peer review of the claim*
+
+#. **Testing:**
+
+   #. Must have system tests which cover all items in the "feature" documentation
+   #. Code Coverage:
+
+      #. *Must have code coverage tracking for the package*
```

Comment:
Propose to have an acceptance criteria of at least 75% code lines coverage for unit tests at this level

Reply:
Similar to this comment: https://github.com/ros-infrastructure/rep/pull/218#discussion_r393888000

Except that in addition to avoiding a specific number we didn't want to have a requirement to maintain a certain level for this quality level. Instead we thought it was sufficient to just have tracking at this level.

I'd like to hear more of why you think this is important and more from others on this issue.